pimd/pim_iface.c

```diff
@@ -1547,27 +1547,20 @@ int pim_if_connected_to_source(struct interface *ifp, struct in_addr src)
 	return 0;
 }
 
-int pim_if_is_loopback(struct pim_instance *pim, struct interface *ifp)
+bool pim_if_is_loopback(struct interface *ifp)
 {
-	if (if_is_loopback(ifp))
-		return 1;
-
-	if (strcmp(ifp->name, pim->vrf->name) == 0)
-		return 1;
+	if (if_is_loopback(ifp) || if_is_vrf(ifp))
```

Comment:
It's been a while since I last looked at pim code, I assume if_is_vrf() checks/matches  the pim vrf name  with the interface like the its replacing above.

Reply:
no if_is_vrf uses the flags passed up from zebra that say the interface is a vrf.

Reply:
ah, thanks!